modules/apps/foundation/frontend-taglib/frontend-taglib-chart/src/main/java/com/liferay/frontend/taglib/chart/model/geomap/GeomapConfig.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.chart.model.geomap;
+
+import com.liferay.frontend.taglib.chart.model.ChartObject;
+
+/**
+ * @author Julien Castelain
+ */
+public class GeomapConfig extends ChartObject {
+
+	public GeomapColor getColor() {
+		return get("color", GeomapColor.class);
+	}
+
+	public Object getData() {
+		return get("data", Object.class);
+	}
+
+	public void setColor(GeomapColor color) {
+		set("color", color);
+	}
+
+	public void setData(Object data) {
```

Comment:
Why is `data` of type `Object`? Isn't it possible to use `String` instead? Or otherwise, pass an `Object` and JSONify it, for example... 

Is it really polymorphic? What can we pass in `data`?

Reply:
@izaera, on the JavaScript side `data` can be a `String`, `Promise` or an `Object`

Since we're not going to deal with `Promise` in Java, I think using `String` would be fine.

If one wants to use an `Object` he could `JSONifyIt` before calling the method.

What do you think? Is this a correct approach?

Reply:
/cc @jbalsas - Thought you might want to double check in case I'm wrong.

Reply:
Keep in mind that the different types produce different results:
- String -> fetch call to the service url identified by the string
- Object -> already resolved data